bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -1441,6 +1446,58 @@ private void sendTextToSpeech() {
         }
     }
 
+    public void textCommand(Device device, String text, @Nullable Integer ttsVolume, @Nullable Integer standardVolume) {
+        if (text.replaceAll("<.+?>", "").replaceAll("\\s+", " ").trim().isEmpty()) {
+            return;
+        }
+
+        // we lock TextCommands until we have finished adding this one
+        Lock lock = locks.computeIfAbsent(TimerType.TEXT_COMMAND, k -> new ReentrantLock());
+        lock.lock();
+        try {
+            TextCommand textCommand = Objects
+                    .requireNonNull(textCommands.computeIfAbsent(Objects.hash(text), k -> new TextCommand(text)));
+            textCommand.devices.add(device);
+            textCommand.ttsVolumes.add(ttsVolume);
+            textCommand.standardVolumes.add(standardVolume);
+            // schedule a TextCommand only if it has not been scheduled before
+            timers.computeIfAbsent(TimerType.TEXT_COMMAND,
+                    k -> scheduler.schedule(this::sendTextCommand, 500, TimeUnit.MILLISECONDS));
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    private synchronized void sendTextCommand() {
+        // we lock new TTS until we have dispatched everything
+        Lock lock = locks.computeIfAbsent(TimerType.TEXT_COMMAND, k -> new ReentrantLock());
+        lock.lock();
+
+        try {
+            Iterator<TextCommand> iterator = textCommands.values().iterator();
+            while (iterator.hasNext()) {
+                TextCommand textCommand = iterator.next();
+                try {
+                    List<Device> devices = textCommand.devices;
+                    if (!devices.isEmpty()) {
+                        String text = textCommand.text;
+                        Map<String, Object> parameters = new HashMap<>();
+                        parameters.put("text", text);
+                        executeSequenceCommandWithVolume(devices, "Alexa.TextCommand", parameters,
```

Comment:
Would this be a good place to use `Map.of`? Or do you need the map to be a HashMap instance?